directord/tests/test_drivers_grpcd.py

```diff
@@ -0,0 +1,846 @@
+        mock_data.command = "command"
+        mock_data.data = "{}"
+        mock_data.info = "info"
+        mock_data.stdout = "stdout"
+        mock_data.stderr = "stderr"
+
+        self.driver._client.get_job.return_value = ("target", mock_data)
+
+        self.driver.job_check = mock_job_check
+
+        data = self.driver.job_recv()
+
+        self.assertEqual(
+            data,
+            [
+                "msg_id",
+                "control",
+                "command",
+                "{}",
+                "info",
+                "stderr",
+                "stdout",
+            ],
+        )
+
+    @mock.patch("directord.utils.get_uuid", return_value="uuid")
+    def test_hearbeat_send(self, mock_uuid):
+        mock_job_send = mock.MagicMock()
+        self.driver.job_send = mock_job_send
+        self.driver.machine_id = "machine_id"
+        self.driver.heartbeat_send(1, 1, 1, "grpcd")
+        mock_job_send.assert_called_once_with(
+            target="DIRECTORD_SERVER",
+            identity="test-node",
+            control="\x05",
+            msg_id="uuid",
+            data=(
+                '{"job_id": "uuid", "version": 1, "host_uptime": 1, '
+                '"agent_uptime": 1, "machine_id": "machine_id", '
+                '"driver": "grpcd"}'
+            ),
+        )
+
+
+class TestGrpcQueueBase(unittest.TestCase):
+    """Test queue base."""
+
+    def setUp(self):
+        grpcd.QueueBase._instance = None
+        self.queue = grpcd.QueueBase.instance()
+
+    def tearDown(self):
+        grpcd.QueueBase._instance = None
+
+    def test_instance(self):
+        """Test instance setup."""
+        self.assertEqual(self.queue._data_queue, {})
+
+    def test_children_queues(self):
+        """Test queue subsclasses."""
+        msg_q = grpcd.MessageQueue.instance()
+        job_q = grpcd.JobQueue.instance()
+        self.assertNotEqual(msg_q, job_q)
+
+    def test_add(self):
+        """Test queue add."""
+        self.queue.add_queue("foo", "bar")
+        self.assertEqual(self.queue._data_queue.get("foo").get(), "bar")
+
+    def test_get(self):
+        """Test queue get."""
+        self.assertEqual(self.queue.get_from_queue("bar"), None)
+        self.queue.add_queue("foo", "bar")
+        self.assertEqual(self.queue.get_from_queue("foo"), "bar")
+
+    def test_check(self):
+        """Test queue check."""
+        self.assertFalse(self.queue.check_queue("foo"))
+        self.queue.add_queue("foo", "bar")
+        self.assertTrue(self.queue.check_queue("foo"))
+        self.queue.get_from_queue("foo")
+        self.assertFalse(self.queue.check_queue("foo"))
+
+    def test_stats(self):
+        """Test queue stats."""
+        self.assertEqual(self.queue.get_stats(), {"targets": []})
+        self.queue.add_queue("foo", "bar")
+        self.assertEqual(self.queue.get_stats(), {"targets": ["foo"]})
+
+    def test_purge(self):
+        """Test queue purge."""
+        self.queue.add_queue("foo", "bar")
```

Comment:
please also add baz, to make sure multiple records get purged as well

Reply:
I mean we just set the entire _data_queue = {} so it doesn't matter whats in there. This doesn't purge a single target's queue which that would test. it resets everything